src/t_cose_encrypt_dec.c

```diff
@@ -0,0 +1,513 @@
+ */
+#include "t_cose/t_cose_encrypt_dec.h"
+#include "t_cose_standard_constants.h"
+#include "psa/crypto.h"
+#include "qcbor/qcbor.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include "qcbor/qcbor_spiffy_decode.h"
+#include <mbedtls/aes.h>
+#include <mbedtls/nist_kw.h>
+#include <mbedtls/hkdf.h>
+#include "t_cose_crypto.h"
+#include "mbedtls/hpke.h"
+
+enum t_cose_err_t
+t_cose_encrypt_dec(struct t_cose_encrypt_dec_ctx* me,
+                   uint8_t *cose,
+                   size_t cose_len,
+                   uint8_t *detached_ciphertext,
+                   size_t detached_ciphertext_len,
+                   uint8_t *plaintext,
+                   size_t plaintext_len,
+                   size_t *plaintext_output_len)
+{
+    QCBORItem              protected_hdr;
+    QCBORItem              inner_protected_hdr;
+    UsefulBufC             nonce_cbor;
+    UsefulBufC             kid_cbor;
+    int64_t                algorithm_id=0;
+    QCBORDecodeContext     DC, DC2;
+    QCBORItem              Item;
+    QCBORItem              Cipher;
+    QCBORError             result;
+    size_t                 key_bitlen;
+    int64_t                alg=0;
+    uint8_t*               ciphertext;
+    size_t                 ciphertext_len;
+
+    int64_t                kty;
+    int64_t                crv;
+    UsefulBufC             peer_key_x;
+    UsefulBufC             peer_key_y;
+
+    /* Temporary storge area */
+    uint8_t                tmp[50];
+    /* Temporary storge area for encrypted cek. */
+    uint8_t                tmp2[50];
+    UsefulBufC             ephemeral={(uint8_t *) tmp,sizeof(tmp)};
+    UsefulBufC             cek_encrypted={(uint8_t *) tmp2,sizeof(tmp2)};
+    size_t                 peer_key_buf_len=0;
+    /*  Temporary storge area for encrypted cek. */
+    uint8_t                peer_key_buf[PSA_EXPORT_PUBLIC_KEY_MAX_SIZE]={0x04};
+    int                    hpke_mode;
+    psa_status_t           status;
+    int                    ret;
+    uint8_t                add_data[20];
+    size_t                 add_data_len=sizeof(add_data);
+    struct q_useful_buf    add_data_struct={add_data,add_data_len};
+    UsefulBufC             add_data_buf;
+    QCBOREncodeContext     additional_data;
+    hpke_suite_t           suite;
+    psa_algorithm_t        psa_algorithm;
+    psa_key_type_t         psa_keytype;
+    bool                   detached_mode;
+    uint8_t                cek[PSA_CIPHER_ENCRYPT_OUTPUT_MAX_SIZE(T_COSE_ENCRYPTION_MAX_KEY_LENGTH)];
+    size_t                 cek_len=PSA_CIPHER_ENCRYPT_OUTPUT_MAX_SIZE(T_COSE_ENCRYPTION_MAX_KEY_LENGTH);
+    psa_key_attributes_t   attributes=PSA_KEY_ATTRIBUTES_INIT;
+    psa_key_handle_t       key_handle=0;
+
+    /* Initialize decoder */
+    QCBORDecode_Init(&DC,
+                     (UsefulBufC){cose,cose_len},
+                     QCBOR_DECODE_MODE_NORMAL);
+
+   /* Make sure the first item is a tag */
+    result=QCBORDecode_GetNext(&DC,&Item);
```

Comment:
The `=` operator should have a space, for consistency with the rest of the library. This comment applies throughout the rest of the source.